airflow/models/taskinstance.py

```diff
@@ -2796,20 +2796,33 @@ def _run_raw_task(
             session=session,
         )
 
-    def _register_asset_changes(self, *, events: OutletEventAccessors, session: Session) -> None:
+    def _register_asset_changes(
+        self, *, events: OutletEventAccessors, session: Session | None = None
+    ) -> None:
+        if session:
+            TaskInstance._register_asset_changes_int(ti=self, events=events, session=session)
+        else:
+            TaskInstance._register_asset_changes_int(ti=self, events=events)
+
+    @staticmethod
+    @internal_api_call
+    @provide_session
+    def _register_asset_changes_int(
+        ti: TaskInstance, *, events: OutletEventAccessors, session: Session = NEW_SESSION
+    ) -> None:
         if TYPE_CHECKING:
-            assert self.task
+            assert ti.task
```

Comment:
Why do we need to use ti instead of self?

Reply:
The internal API (in AIP-44) can not call on class instances, can only use static methods. That's why I needed to add a wrapper and switch to `ti` instead of `self`.